sqlserver/datadog_checks/sqlserver/queries.py

```diff
@@ -156,3 +156,53 @@
         {'name': 'failover_cluster', 'type': 'tag'},
     ],
 }
+
+_file_stats_column_name_to_metric_name = {
+    'size_on_disk_bytes': 'files.size_on_disk',
+    'num_of_reads': 'files.reads',
+    'num_of_bytes_read': 'files.read_bytes',
+    'io_stall_read_ms': 'files.read_io_stall',
+    'io_stall_queued_read_ms': 'files.read_io_stall_queued',
+    'num_of_writes': 'files.writes',
+    'num_of_bytes_written': 'files.written_bytes',
+    'io_stall_write_ms': 'files.write_io_stall',
+    'io_stall_queued_write_ms': 'files.write_io_stall_queued',
+    'io_stall': 'files.io_stall',
+}
+
+
+def get_query_file_stats(available_file_stats_columns):
```

Comment:
Why not just declare two queries, which would be much simpler:

```python
QUERY_FILE_STATS = """..."""

QUERY_FILE_STATS_LTE_2012 = """..."""
```

Reply:
Updated to use the version to decide which columns are available instead of having to query the table to get the full set of available columns. 

I think dynamically constructing the query in a single function is still simpler overall than copy-pasting the whole thing twice.